src/mappings/eraManager.ts

```diff
@@ -19,16 +19,23 @@ export async function handleNewEra(
   if (id.gt(1)) {
     const previousId = id.sub(1);
     const previousEra = await Era.get(previousId.toHexString());
-    assert(previousEra, `Era ${previousId.toNumber()} doesn't exist`);
-
-    previousEra.endTime = event.blockTimestamp;
-
-    await previousEra.save();
+    if (previousEra) {
+      previousEra.endTime = event.blockTimestamp;
+      await previousEra.save();
+    } else {
+      const previousEra = Era.create({
+        id: previousId.toHexString(),
+        startTime: event.blockTimestamp,
```

Comment:
So for startTime, may be good to calculate with eraPeriod:

```
  const eraManager = EraManager__factory.connect(
    ERA_MANAGER_ADDRESS,
    new FrontierEthProvider()
  );

const eraPeroid = await eraManager.eraPeriod();
const startTime = event.blockTimestamp - eraPeroid.toNumber();
const endTime =  event.blockTimestamp;
```

Reply:
Good point! Updated, and thanks @HZHAIX